test/e2e/cluster_upgrade_runtimesdk.go

```diff
@@ -464,23 +438,22 @@ func runtimeHookTestHandler(ctx context.Context, c client.Client, namespace, clu
 		if err := checkLifecycleHooksCalledAtLeastOnce(ctx, c, namespace, clusterName, []string{hookName}); err != nil {
 			return err
 		}
-		cluster := &clusterv1.Cluster{}
-		if err := c.Get(ctx, client.ObjectKey{Namespace: namespace, Name: clusterName}, cluster); err != nil {
-			return err
-		}
 
 		// Check for the existence of the condition if withTopologyReconciledCondition is true.
-		if withTopologyReconciledCondition &&
-			(conditions.GetReason(cluster, clusterv1.TopologyReconciledCondition) != clusterv1.TopologyReconcileFailedReason) {
-			return errors.New("Condition not found on Cluster object")
+		if withTopologyReconciledCondition {
+			cluster := framework.GetClusterByName(ctx, framework.GetClusterByNameInput{Name: clusterName, Namespace: namespace, Getter: c})
+
+			if !clusterConditionShowsHookBlocking(cluster, hookName) {
+				return errors.Errorf("Blocking condition for %s not found on Cluster object", hookName)
+			}
 		}
 		return nil
-	}, 60*time.Second).Should(Succeed(), "%s has not been called", hookName)
+	}, 30*time.Second).Should(Succeed(), "%s has not been called", hookName)
 
 	// blockingCondition should consistently be true as the Runtime hook is returning "Failure".
 	Consistently(func() bool {
 		return blockingCondition()
-	}, 30*time.Second).Should(BeTrue(),
+	}, 60*time.Second).Should(BeTrue(),
```

Comment:
Do we really need to wait 1 minute to confirm we are blocked? 30 seems enough to me...

Reply:
For now this is mainly up to a minute because the condition test doesn't work for beforeClusterUpgrade (working on fixing this in this PR or a follow up) once that's fixed I think 30 seconds is enough, but and upgrade specifically could take long enough to kick in that I'm more comfortable with 60 seconds with the current test structure.